diff --git a/plugins/export_ui/jw_player_ctools_export_ui.inc b/plugins/export_ui/jw_player_ctools_export_ui.inc
index a57f315..7659e9c 100644
--- a/plugins/export_ui/jw_player_ctools_export_ui.inc
+++ b/plugins/export_ui/jw_player_ctools_export_ui.inc
@@ -52,7 +52,7 @@ function jw_player_ctools_export_ui_form(&$form, &$form_state) {
     '#description' => t('Enter name for the preet.'),
     '#default_value' =>  $preset->preset_name,
     '#required' => true,
-    '#weight' => 0
+    '#weight' => 0,
   );
 
   $form['info']['machine_name'] = array(
@@ -80,7 +80,7 @@ function jw_player_ctools_export_ui_form(&$form, &$form_state) {
     '#field_suffix' => ' ' . t('pixels'),
     '#default_value' => $settings['width'],
     '#required' => true,
-    '#weight' => 5
+    '#weight' => 5,
   );
 
   $form['settings']['height'] = array(
@@ -91,9 +91,52 @@ function jw_player_ctools_export_ui_form(&$form, &$form_state) {
     '#field_suffix' => ' ' . t('pixels'),
     '#default_value' => $settings['height'],
     '#required' => true,
-    '#weight' => 6
+    '#weight' => 6,
   );
 
+  $form['settings']['controlbar'] = array(
+    '#title' => t('Controlbar Position'),
+    '#type' => 'select',
+    '#description' => t('The positon of the controlbar'),
+    '#default_value' => !empty($settings['controlbar']) ? $settings['controlbar']: 'none',
+    '#options' => array(
+      'none' => t('None'),
+      'bottom' => t('Bottom'),
+      'top' => t('Top'),
+      'over' => t('Over')
+    ),
+    '#weight' => 7,
+  );
+
+  // playlist settings
+  $form['settings']['playlist.position'] = array(
+    '#title' => t('Playlist Position'),
+    '#type' => 'select',
+    '#description' => t('The positon of the playlist'),
+    '#default_value' => !empty($settings['playlist.position']) ? $settings['playlist.position']: 'right',
+    '#options' => array(
+      'none' => t('None'),
+      'bottom' => t('Bottom'),
+      'top' => t('Top'),
+      'right' => t('Right'),
+      'left' => t('Left'),
+      'over' => t('Over')
+    ),
+    '#weight' => 8,
+  );
+
+   $form['settings']['playlist.size'] = array(
+    '#title' => t('Playlist Width'),
+    '#type' => 'textfield',
+    '#description' => t('The playlist width in pixels.'),
+    '#default_value' => !empty($settings['playlist.size']) ? $settings['playlist.size']: '180',
+    '#field_suffix' => ' ' . t('pixels'),
+    '#size' => 10,
+    '#weight' => 9,
+  );
+
+
+  // skins
   $skin_options = array();
   foreach (jw_player_skins() as $skin) {
     $skin_options[$skin->name] = ucfirst($skin->name);
@@ -128,4 +171,4 @@ function jw_player_ctools_export_ui_form_validate($form, &$form_state) {
   if (!is_numeric($vals['settings']['height'])) {
     form_set_error('width', 'Only numeric values allowed for height' );
   }
-}
\ No newline at end of file
+}
