diff --git a/jw_player.js b/jw_player.js
index 3f863ff..570a8db 100644
--- a/jw_player.js
+++ b/jw_player.js
@@ -4,17 +4,20 @@
    * Attaches a JW Player element by using the JW Player Javascript Library
    */
   Drupal.behaviors.JWPlayer = {
-    attach: function(context, settings) {
-      $.each(settings.jw_player, function(player_id, config) {
-        jwplayer(player_id).setup(config);
+    attach: function(context) {
+      var players = Drupal.settings.jw_player;
+      $.each(players, function(player_id, config) {
+        if ($('#' + player_id, context).length) {
+          jwplayer(player_id).setup(config);
 
-        if (config.events) {
-          $.each(config.events, function(event, callback) {
-            jwplayer(player_id)[event](eval(callback));
-          });
-        };
+          if (config.events) {
+            $.each(config.events, function(event, callback) {
+              jwplayer(player_id)[event](eval(callback));
+            });
+          };
+        }
       });
     }
   };
 
-})(jQuery);
\ No newline at end of file
+})(jQuery);
